### DISCUSSION TITLE
Assistance Needed: Resolving "Error HH18" with Hardhat in Termux Environment

### DISCUSSION THREAD

**Discussion by TechJosh89** [NomicFoundation/hardhat#4937] [Troubleshooting]
Dear Hardhat Community,

I hope this message finds you well. I am reaching out for assistance with a persistent issue I have encountered while setting up and running Hardhat in a Termux environment on a Samsung Galaxy Tab S9. Despite following the standard installation and setup procedures and consulting ChatGPT, I consistently face "Error HH18" related to a corrupted lockfile due to NPM bug #4828, even after attempting various troubleshooting steps.

Here is a brief overview of my setup and the steps I have taken:

Environment: Termux on Samsung Galaxy Tab S9
Node.js Version: 21.6.2 (confirmed as current and supported by Hardhat)
Initial Setup: I initiated a new npm project, installed nodejs packages, and then proceeded to set up Hardhat using npm install --save-dev hardhat.
Issue Encountered: Upon running npx hardhat test, I encountered "Error HH18," suggesting that Hardhat was installed with a corrupted lockfile due to NPM bug #4828. The suggested solution is to delete node_modules and package-lock.json, then reinstall the project, which I've done multiple times to no avail.
Troubleshooting Steps:
-Cleared npm cache and reinstalled packages
-Attempted installation with Yarn, facing the same issue
-Ensured all dependencies and environment variables (including Python for node-gyp) were correctly set
-Reinitialized the Hardhat project in a new directory

Despite these efforts, the "Error HH18" persists, preventing me from progressing with my Hardhat project.

I would greatly appreciate any insights, suggestions, or guidance on how to resolve this issue. Has anyone encountered a similar problem, particularly in a Termux or tablet-based development environment? Any advice on troubleshooting steps that might have been overlooked or alternative approaches to setting up Hardhat would be invaluable.

Thank you for your time and assistance. I look forward to any feedback or suggestions you may offer.

Best regards,

Josh

**Comment by kanej** [comment]
Hello, 

Our initial suspicion is that the the platform isn't supported or we are misidentifying the platform in `@nomicfoundation/solidity-analyzer`. The analyzer is built and rust and has platform specific builds. Due to an unrelated `npm` issue we try and detect when we are on a platform that differs from the one the npm lock file was first created on. Our suspicion is that that check is masking the underlying problem that the Galaxy Tab S9 is not a supported platform for our solidity analysis library, or we are misidentifying its platform.

The platforms currently supported are:

![image](https://github.com/NomicFoundation/hardhat/assets/24030/3e263f59-fae7-460e-b3e6-f66f4fb80995)

Is the Galaxy Tab S9 a platform on that list?

Could I ask you to try and install the solidity analyzer itself, the the [minimal analyze sample from the readme](https://www.npmjs.com/package/@nomicfoundation/solidity-analyzer#example): https://www.npmjs.com/package/@nomicfoundation/solidity-analyzer

**Reply by ArmaanSoniQG** [reply]
I am facing the same problem as Josh and I am on win32-x64-msvc. Tried both npm and yarn and cached everything, same as him.
